### PR TITLE
More augmentation of debugging code

### PR DESCRIPTION
**Pull request by steven-johnson**
This expands on #6182 by added tracking for BoxesTouched, and integrating the nesting levels with the previous code. This allows a more complete vision of what's happening during bounds calculation.

### CODE REVIEW

File: src/Bounds.cpp [hunk 1]
```diff
@@ -1984,6 +1966,133 @@ class BoxesTouched : public IRGraphVisitor {
...
+        }
+
+        ~BoxesTouchedLogger() {
+            log_box_diffs(this->boxes, self->boxes);
+            self->log_indent--;
+            log_line('}');
+            internal_assert(self->current_logger == this);
+            if (parent_logger) {
+                // Propagate changes to our parent.
+                // This isn't efficient at all, but it's ususally-disabled debugging code.
```
Comment (abadams) [inline src/Bounds.cpp:2059]: ususally

File: src/Bounds.cpp [hunk 2]
```diff
@@ -2845,7 +2967,7 @@ map<string, Box> boxes_touched(const Expr &e, Stmt s, bool consider_calls, bool
             debug(0) << "  " << j << ": " << it.second[j].min
                      << " .. "
                      << it.second[j].max
-                     << "\n";
+                     << '\n';
```
Comment (abadams) [inline src/Bounds.cpp:2970]: I dug into benchmarking `<< '\n'` vs `<< "\n"` at one time, and weirdly enough, `<< "\n"` is faster. Sending a single character to a stream goes through a weird slow path.

Comment (abadams) [reply src/Bounds.cpp:2970]: (in libstdc++)

File: src/Bounds.h [hunk 3]
```diff
@@ -33,7 +33,8 @@ const FuncValueBounds &empty_func_value_bounds();
 Interval bounds_of_expr_in_scope(const Expr &expr,
                                  const Scope<Interval> &scope,
                                  const FuncValueBounds &func_bounds = empty_func_value_bounds(),
-                                 bool const_bound = false);
+                                 bool const_bound = false,
+                                 int indent = 0);
```
Comment (abadams) [inline src/Bounds.h:37]: Is this indent parameter only used internally? I don't think we should expose it.

File: src/Bounds.cpp (hunk 1 above)
Comment (steven-johnson) [reply src/Bounds.cpp]: Done

File: src/Bounds.cpp (hunk 2 above)
Comment (steven-johnson) [reply src/Bounds.cpp]: Done

File: src/Bounds.h (hunk 3 above)
Comment (steven-johnson) [reply src/Bounds.h:37]: Yep. I just felt like the alternative (making an internal-only wrapper) was even uglier. Let me look again.

Comment (steven-johnson) [reply src/Bounds.h]: PTAL

**Review by abadams — APPROVED** [review]

**Pull request merged**